src/gluonts/itertools.py

```diff
@@ -193,7 +193,7 @@ def __len__(self):
         return len(self.iterable)
 
     def __repr__(self):
-        return f"Map(data={self.iterable!r})"
+        return f"Map(iterable={self.iterable!r})"
```

Comment:
How about the `fn` argument though?

Reply:
I guess we could also add that. Then let's just make this a dataclass.